src/modules/transaction/store/actions.js

```diff
@@ -115,27 +116,37 @@ export const transactionBroadcasted =
       // https://github.com/LiskHQ/lisk-desktop/issues/4698 should handle this logic
     }
 
-    if (dryRunResult.data?.result === -1) {
+    if (error) {
       dispatch({
         type: actionTypes.broadcastedTransactionError,
         data: {
-          error: dryRunResult.data?.errorMessage,
+          error: error.message,
           transaction,
         },
       });
-    }
+    } else {
+      if (dryRunResult.data?.result === -1) {
+        dispatch({
+          type: actionTypes.broadcastedTransactionError,
+          data: {
+            error: dryRunResult.data?.errorMessage,
+            transaction,
+          },
+        });
+      }
 
-    if (dryRunResult.data?.result === 0) {
-      // @TODO: Prepare error message by parsing the events based on each transaction type
-      // https://github.com/LiskHQ/lisk-desktop/issues/4698 should resolve all the dry run related logic along with feedback
-      const temporaryError = dryRunResult.data?.events.map((e) => e.name).join(', ');
-      dispatch({
-        type: actionTypes.broadcastedTransactionError,
-        data: {
-          error: temporaryError,
-          transaction,
-        },
-      });
+      if (dryRunResult.data?.result === 0) {
+        // @TODO: Prepare error message by parsing the events based on each transaction type
+        // https://github.com/LiskHQ/lisk-desktop/issues/4698 should resolve all the dry run related logic along with feedback
+        const temporaryError = dryRunResult.data?.events.map((e) => e.name).join(', ');
+        dispatch({
+          type: actionTypes.broadcastedTransactionError,
+          data: {
+            error: temporaryError,
+            transaction,
```

Comment:
In all 3 cases, error/0/-1 only the `error` is different? so why not remove the duplication and just prepare the error and dispatch?

e.g 

```
const transactionError = error || dryRunResult.data?.result === 0 ? dryRunResult.data?.events.map((e) => e.name).join(', ') : dryRunResult.data?.errorMessage;

dispatch({
          type: actionTypes.broadcastedTransactionError,
          data: {
            error: transactionError,
            transaction,
          },
        });
```

Reply:
The snippet you provided will always return an error message while the old code sometimes did not dispatch anything.  Are we sure that we want to make this change? According to previous code the function can in some cases never dispatch anything but return just false. With the snippet in your comment we would always dispatch something, maybe that is what we want?


To make it behave like before we would need something like this 
![image](https://github.com/LiskHQ/lisk-desktop/assets/8784876/dde3bde2-0d20-4894-b629-d0341f9c13cc)

Reply:
Ideally if the API or dry run is not successful we should always dispatch an error to abort the operation, so I think the changes you made should be fine?